modules/ting/ting.module

```diff
@@ -468,20 +452,26 @@ function ting_ding_entity_buttons($type, $entity) {
       );
     }
 
-    $collection = ting_collection_load($entity->id);
-    if (count($collection->types) > 1) {
-      $buttons[] = array(
-        '#theme' => 'link',
-        '#text' => t('Other formats'),
-        '#path' => '#',
-        '#options' => array(
-          'attributes' => array(
-            'class' => array('action-button', 'other-formats'),
-          ),
-          'html' => FALSE,
-          'external' => TRUE,
-        ),
-      );
+    // Loading collection (specially on empty cache is expensive). So only load
+    // the collection when it's need and here only when this button is required
+    // in the design. Before using context it was load but not used in the
+    // design.
+    if (strpos($view_mode, 'teaser') === FALSE) {
+      $collection = ting_collection_load($entity->id);
+      if (count($collection->getTypes()) > 1) {
+        $buttons[] = [
+          '#theme' => 'link',
+          '#text' => t('Other formats'),
+          '#path' => '#',
+          '#options' => [
+            'attributes' => [
+              'class' => ['action-button', 'other-formats'],
```

Comment:
To be honest such a button could as easily be generated clientside. It has no relevance serverside.

Reply:
It all comes back to the way the entity_button hook is miss-used in the theme.

E.g. https://github.com/ding2/ding2/blob/master/themes/ddbasic/template.php#L825